DVCSMP-1803 Springs Window Fashions Integration

greens opened the pull request:
This adds two DTHs, one for generic window shades, and one specificaly for SWF. The SWF handler has a couple differences:

* Battery is displayed as a binary state: green/good or red/bad
* The presetPosition function is handled on the device hardware
* The direction of commands can be reversed to reflect changes to the configuration of the motor

greens committed:
Springs Window Fashions Integration

This adds two DTHs, one for generic window shades, and one specificaly for SWF. The SWF handler has a couple differences:

* Battery is displayed as a binary state: green/good or red/bad
* The presetPosition function is handled on the device hardware
* The direction of commands can be reversed to reflect changes to the configuration of the motor

workingmonk commented on devicetypes/smartthings/springs-window-fashions-shade.src/springs-window-fashions-shade.groovy:
```diff
@@ -0,0 +1,267 @@
+/**
+ *  Copyright 2016 SmartThings
```
2017

workingmonk commented on devicetypes/smartthings/zwave-window-shade.src/zwave-window-shade.groovy:
```diff
@@ -0,0 +1,248 @@
+/**
+ *  Copyright 2016 SmartThings
```
2017

workingmonk commented:
Will wait for @mckeed to approve this, but do not merge until cert is complete for the device. New DTHs with fingerprints first go through cert and then merged into the repo

greens commented:
Thanks for the info.

Just wanted to get eyes on it sooner rather than later.

mckeed commented on devicetypes/smartthings/springs-window-fashions-shade.src/springs-window-fashions-shade.groovy:
```diff
@@ -0,0 +1,267 @@
+
+def presetPosition() {
+    // SWF blinds handle this data independently
+    if (device.getDataValue("MSR")?.startsWith("026E")) {
```
We can just take out all the state.preset stuff for this one.

greens replied:
done

greens committed:
DVCSMP-1803 Window Blinds Integration

Updates copyright and adds new icons.

greens committed:
DVCSMP-1803 Springs Window Fashions Integrations

0xFF does not test as >99, thanks signed integers.

Ironically, had some logic around direction reversing reversed.

greens committed:
DVCSMP-1803 Window Blinds integration

Removing 'switch' capability as it's redundant with the window shade/switch level combination.

greens committed:
DVCSMP-1803 Window Blinds integration

Removing 'switch' capability as it's redundant with the window shade/switch level combination.

greens committed:
ICP-3152 "Battery" is cut off in app

Changing the "battery" string to "batt." Ideal solution would be a battery icon.

mckeed commented on devicetypes/smartthings/springs-window-fashions-shade.src/springs-window-fashions-shade.groovy:
```diff
@@ -72,7 +72,7 @@ metadata {
 
         valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
-            state "battery", label:'battery', unit:"",
+            state "battery", label:'batt.', unit:"",
```
Is there also a ticket opened for the app cutting off the text? That probably affects a bunch of devices, right?

greens replied:
ICP-3152

greens replied:
That's just for this one. Don't know about a "general" one. I mentioned in the commit message that the ideal solution would just be a battery icon, but we actually don't have one.

greens committed:
DVCSMP-1803 Springs Window Fashions integration

Remove manufacturer-specific code from SWF-specific DTH

dkirker commented:
@greens Hey, so I just noticed that the "partiallyOpen" state value for windowShade doesn't appear to follow the capability spec. It looks like it should be "partially open"? See the capabilities repository.

greens committed:
DVCSMP-1803 Springs Window Fashions integration

Update "partially open" value to match Capability

greens commented:
@dkirker good catch. Fixed.

greens committed:
DVCSMP-1803 Window covering integration

Add the ocf device type

greens committed:
DVCSMP-1803 Window covering integration

Small hacky fix to map 99 to 100% open.

greens committed:
ICP-3603 Window Shades remain offline

Changed these devices to be offline pingable. Lowered check interval to 4 hours.

greens committed:
ICP-3604 Spring window remote

Creates a new "blank" DTH for the springs remotes.

greens committed:
ICP-3604 Spring window remote

Add an OCF device type for fun.

greens committed:
Hide the remotes from the device list.

greens committed:
DVCSMP-3760 Erroneous low battery reports

Springs devices will sometimes report a low battery when they are rapidly manually actuated. This will drop those events on the floor to limit the bad data getting propagated.

greens committed:
Merge branch 'master' into SWF

greens committed:
Merge branch 'master' into SWF

dkirker commented on devicetypes/smartthings/springs-window-fashions-remote.src/springs-window-fashions-remote.groovy:
```diff
@@ -0,0 +1,115 @@
+        }
+
+        valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+            state "battery", label:'batt.', unit:"",
```
@greens I keep forgetting to leave this comment, and am surprised that QA hasn't said anything about it, but do we want this to display "batt." or the actual percentage?

dkirker replied:
Ok, ignore this, since I just talked to you and it's all sub optimal. :/

dkirker commented on devicetypes/smartthings/springs-window-fashions-shade.src/springs-window-fashions-shade.groovy:
```diff
@@ -0,0 +1,259 @@
+        }
+
+        valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+            state "battery", label:'batt.', unit:"",
```
Same here as above.
